test/uk/gov/hmrc/economiccrimelevyregistration/services/RegistrationValidationServiceSpec.scala

```diff
@@ -375,5 +372,39 @@ class RegistrationValidationServiceSpec extends SpecBase {
           )
         )
     }
+
+    "return errors if the registration for a other entity is invalid" in forAll {
+      (validCharityRegistration: ValidCharityRegistration) =>
+        val invalidRegistration = validCharityRegistration.registration.copy(
+          optOtherEntityJourneyData = None
+        )
+        val result              = service.validateRegistration(invalidRegistration)
+        result.isValid shouldBe false
+        result.leftMap(nec => nec.toList.isEmpty shouldBe false)
+    }
```

Comment:
```suggestion
       "return errors if the registration for a other entity is invalid" in forAll { internalId: String =>
      val emptyOtherRegistration = Registration.empty(internalId).copy(entityType = Some(Other))

      val result = service.validateRegistration(emptyOtherRegistration)

      val expectedErrors = Seq(
        DataValidationError(DataMissing, "Carried out AML regulated activity choice is missing"),
        DataValidationError(DataMissing, "Relevant AP 12 months choice is missing"),
        DataValidationError(DataMissing, "Relevant AP revenue is missing"),
        DataValidationError(DataMissing, "Revenue meets threshold flag is missing"),
        DataValidationError(DataMissing, "AML supervisor is missing"),
        DataValidationError(DataMissing, "Business sector is missing"),
        DataValidationError(DataMissing, "First contact name is missing"),
        DataValidationError(DataMissing, "First contact role is missing"),
        DataValidationError(DataMissing, "First contact email is missing"),
        DataValidationError(DataMissing, "First contact number is missing"),
        DataValidationError(DataMissing, "Contact address is missing"),
        DataValidationError(DataMissing, "Second contact choice is missing"),
        DataValidationError(DataMissing, "Other entity type is missing"),
        DataValidationError(DataMissing, "Other entity data is missing")
      )

      result.isValid shouldBe false
      result.leftMap(nec => nec.toList should contain theSameElementsAs expectedErrors)
    }
```

I meant like this - this test also fails like this (TDD!) which highlights something missing in the validation, you have missed `validateEclAddress(registration.contactAddress),` from the other entity validation